src/libraries/System.Private.Uri/src/System/UriSyntax.cs

```diff
@@ -267,7 +269,12 @@ internal UriParser InternalOnNewUri()
 
         internal void InternalValidate(Uri thisUri, out UriFormatException? parsingError)
         {
+            thisUri.DebugAssertInCtor();
             InitializeAndValidate(thisUri, out parsingError);
+
+            // InitializeAndValidate should not be called outside of the constructor
+            Debug.Assert(sizeof(Uri.Flags) == sizeof(ulong));
+            Interlocked.Or(ref Unsafe.As<Uri.Flags, ulong>(ref thisUri._flags), (ulong)Uri.Flags.CustomParser_ParseMinimalAlreadyCalled);
```

Comment:
This part sets the `CustomParser_ParseMinimalAlreadyCalled` flag so that we can catch `InitializeAndValidate` being called outside of the override, even if we never call the base implementation.